tasks/libs/common/gomodules.py

```diff
@@ -0,0 +1,311 @@
+
+        base_dir = base_dir or Path.cwd()
+
+        with open(base_dir / cls.FILE_NAME) as file:
+            data = yaml.safe_load(file)
+
+        return Configuration.from_dict(data)
+
+    def to_dict(self) -> dict[str, object]:
+        modules_config = {}
+        # Path removed because the key is the path
+        modules_config.update(
+            {name: module.to_dict(remove_path=True) or 'default' for name, module in self.modules.items()}
+        )
+        modules_config.update({module: 'ignored' for module in self.ignored_modules})
+
+        return {
+            'modules': modules_config,
+        }
+
+    def to_file(self):
+        """Save the configuration to a yaml file at <base_dir/FILE_NAME>."""
+
+        with open(self.base_dir / self.FILE_NAME, "w") as file:
+            path = f'tasks/{Path(__file__).relative_to(Path(tasks.__file__).parent).as_posix()}'
+            print(self.INFO_COMMENT.format(file=path).strip() + '\n', file=file)
+
+            yaml.dump(self.to_dict(), file, Dumper=ConfigDumper)
+
+
+@dataclass
+class GoModule:
+    """A Go module abstraction.
+
+    See:
+        Documentation can be found in <docs/dev/modules.md>.
+
+    Args:
+        targets: Directories to unit test.
+        condition: When to execute tests, must be a enumerated field of `GoModule.CONDITIONS`.
+        should_tag: Whether this module should be tagged or not.
+        importable: HACK: Workaround for modules that can be tested, but not imported (eg. gohai), because they define a main package A better solution would be to automatically detect if a module contains a main package, at the cost of spending some time parsing the module.
+        independent: specifies whether this modules is supposed to exist independently of the datadog-agent module. If True, a check will run to ensure this is true.
+        lint_targets: Directories to lint.
+        used_by_otel: Whether the module is an otel dependency or not.
+
+    Usage:
+        A module is defined within the modules.yml file containing the following fields by default (these can be omitted if the default value is used):
+        > condition: always
+        > importable: true
+        > independent: true
+        > lint_targets:
+        > - .
+        > should_tag: true
+        > targets:
+        > - .
+        > used_by_otel: false
+
+        If a module has default attributes, it should be defined like this:
+        > my/module: default
+
+        If a module should be ignored and not included within get_default_modules(), it should be defined like this:
+        > my/module: ignored
+    """
+
+    # Possible conditions for GoModule.condition
+    CONDITIONS: ClassVar[dict[str, Callable]] = {
+        'always': lambda: True,
+        'never': lambda: False,
+        'is_linux': lambda: sys.platform == "linux",
+    }
+
+    # Posix path of the module's directory
+    path: str
+    # Directories to unit test
+    targets: list[str] | None = None
```

Comment:
```suggestion
    test_targets: list[str] | None = None
```
If they are target for tests, the name could carry this information, wdyt?